simplq/src/components/pages/PageNotFound/index.jsx

```diff
@@ -1,12 +1,29 @@
 import React from 'react';
+import { isJoinPageUrl } from '../../../utils/textOperations';
 import Button from '../../common/Button';
 import styles from './pageNotFound.module.scss';
+import CreateJoinForm from '../../common/CreateJoinForm';
 
 function PageNotFound(props) {
-  return (
-    <>
+  const MainContent = () => {
+    const urlPath = window.location.href;
+    if (isJoinPageUrl(urlPath)) {
```

Comment:
In Join page, this is what we are doing:

```
  const [error, setError] = useState(false);
  useEffect(() => {
    async function fetchData() {
      const response = await QueueService.getStatusByName(queueName);
      if (response) {
        setQueueStatusResponse(response);
      } else {
        setError(true);
      }
    }
    fetchData();
  }, [queueName]);

  if (error) {
    return <PageNotFound history={props.history} />;
  }
```

Here on error, we are rendering the 404 page (as if it was a component). Instead can we route to 404 page like this? I feel this is more proper:

```
  useEffect(() => {
    async function fetchData() {
      const response = await QueueService.getStatusByName(queueName);
      if (response) {
        setQueueStatusResponse(response);
      } else {
        props.history.push('/queueNotFound/queueName='+queueName)
      }
    }
    fetchData();
  }, [queueName]);
```

Then define a new route which would also render the same 404 page:

```
<Route path="/pageNotFound/queueName=:queueName" exact component={QueueStatus} />
```
The catch all 404 route also stays.

Then you can conditionally check and retrieve the queueName from `props.match.params.queueName` here. Will this work?